### PR TITLE
Properly handle different networks when resolving checkpoints

### PR DESCRIPTION
**Pull request by cburgdorf**
### What was wrong?

As mentioned in #1209, currently when using `--beam-from-checkpoint eth://block/byetherscan/latest` with non-mainnet networks it will silently try to do the wrong thing.

### How was it fixed?

Use the correct Etherscan API for every supported network. Also, tell the user when a network does not support this API.

**Bonus**: There were some leaking exceptions that I turned into a proper user facing error. That's in a separate commit but I sneaked it into this PR.

### To-Do

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: trinity/bootstrap.py [hunk 1]
```diff
@@ -100,10 +102,13 @@ def main_entry(trinity_boot: BootFn,
 
     argcomplete.autocomplete(parser)
 
-    args = parser.parse_args()
+    try:
+        args = parser.parse_args()
+    except ValidationError as exc:
+        parser.error(str(exc))
```
Comment (carver) [inline trinity/bootstrap.py]: Maybe we can get the whole stack trace in here, since an exception that bubbles up this far seems unintended.

Comment (cburgdorf) [reply trinity/bootstrap.py]: >since an exception that bubbles up this far seems unintended.

Oh, I understand! Let me roll this back. I didn't know there exists and actual `argparse.ArgumentError` that should be thrown when a custom argument parsing failed.

Now that I know I went to this other place and made sure it raises an `argsparse.Argumenterror` and then we don't need to catch `ValidationError` here :)

https://github.com/ethereum/trinity/pull/1269/commits/6b8d897aba1a1c2cc0e9febfe012137a0a06547c#diff-4398c8abe4c50ec44530591d4bbc1ee8R113

File: trinity/components/builtin/syncer/cli.py [hunk 2]
```diff
@@ -106,5 +115,9 @@ def __call__(self,
                  namespace: argparse.Namespace,
                  value: Any,
                  option_string: str=None) -> None:
-        parsed = parse_checkpoint_uri(value)
+
+        try:
+            parsed = parse_checkpoint_uri(value, namespace.network_id)
```
Comment (pipermerriam) [inline trinity/components/builtin/syncer/cli.py]: not part of this PR but what if we used our own type for `namespace` instances.

```python
class TrinityNamespace(argparse.Namespace):
    network_id: int
    disable_upnp: Optional[bool]
    ...
```

Then we could at least be explicit about what properties it has and have stronger types in our code that accesses them.

File: trinity/components/builtin/syncer/etherscan_api.py [hunk 3]
```diff
@@ -9,19 +10,37 @@
...
-ETHERSCAN_PROXY_API_URL = f"{ETHERSCAN_API_URL}?module=proxy"
-
-
 class EtherscanAPIError(BaseTrinityError):
     pass
 
 
-def etherscan_post(action: str) -> Any:
-    response = requests.post(f"{ETHERSCAN_PROXY_API_URL}&action={action}")
+class Network(enum.Enum):
```
Comment (pipermerriam) [inline trinity/components/builtin/syncer/etherscan_api.py]: if you use `enum.IntEnum` then you'll be able to do things like `if network_id == Network.MAINNET`
